core/server/DatabaseAPI.cpp

```diff
@@ -3198,8 +3206,14 @@ void DatabaseAPI::taskInitDataPut(const Auth::DataPutRequest &a_request,
     payload["check"] = a_request.check();
   }
 
-  string body = payload.dump(-1, ' ', true);
+  if (a_request.has_collection_id()) {
+    payload["collection_id"] = a_request.collection_id();
+  }
+  if (a_request.has_collection_type()) {
+    payload["collection_type"] = a_request.collection_type();
+  }
 
+  string body = payload.dump(-1, ' ', true);

```

Comment:
Nice catch, it could be useful to add comments in both requests where this is used to incidate that this is compacting the json before sending. It's not immediately clear from looking at the code that this is what is happening.